webpack/scenes/Subscriptions/Manifest/CdnConfigurationForm.js

```diff
@@ -0,0 +1,166 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { useDispatch, useSelector } from 'react-redux';
+
+import {
+  ActionGroup,
+  Button,
+  Form,
+  FormGroup,
+  FormSelect,
+  FormSelectOption,
+  Spinner,
+  TextInput,
+} from '@patternfly/react-core';
+
+import { translate as __ } from 'foremanReact/common/I18n';
+
+import EditableTextInput from '../../../components/EditableTextInput';
+
+import {
+  selectUpdatingCdnConfiguration,
+} from '../../Organizations/OrganizationSelectors';
+
+import { updateCdnConfiguration } from '../../Organizations/OrganizationActions';
+import './CdnConfigurationForm.scss';
+
+const CdnConfigurationForm = (props) => {
+  const {
+    contentCredentials,
+    cdnConfiguration,
+  } = props;
+
+  const dispatch = useDispatch();
+
+  const [url, setUrl] = useState(cdnConfiguration.url);
+  const [username, setUsername] = useState(cdnConfiguration.username);
+  const [password, setPassword] = useState(null);
+  const [organizationLabel, setOrganizationLabel] =
+    useState(cdnConfiguration.upstream_organization_label);
+  const [sslCaCredentialId, setSslCaCredentialId] = useState(cdnConfiguration.ssl_ca_credential_id);
+  const updatingCdnConfiguration = useSelector(state => selectUpdatingCdnConfiguration(state));
+
+  const editPassword = (value) => {
+    if (value === null) {
+      setPassword('');
+    } else {
+      setPassword(value);
+    }
+  };
+
+  const hasPassword = (cdnConfiguration.password_exists && password === null)
+    || password?.length > 0;
+
+  const performUpdate = () => {
+    dispatch(updateCdnConfiguration({
+      url,
+      username,
+      password,
+      upstream_organization_label: organizationLabel,
+      ssl_ca_credential_id: sslCaCredentialId,
+    }));
+  };
+
+  const updateButtonContent = (
+    updatingCdnConfiguration ? (
+      <React.Fragment>
+        <Spinner size="md" />
```

Comment:
You can just add an `isLoading` prop to the `<Button>` component and get all this for free.  (and with better spacing between the spinner and text :wink: )